system/HTTP/CURLRequest.php

```diff
@@ -309,7 +309,9 @@ protected function parseOptions(array $options)
 
         if (array_key_exists('headers', $options) && is_array($options['headers'])) {
             foreach ($options['headers'] as $name => $value) {
-                $this->setHeader($name, $value);
+                if (is_scalar($value)) {
```

Comment:
In my opinion if an unacceptable value is passed it should not be quietly skipped but rather should throw an exception. Otherwise we are creating a scenario where developers have to dig pretty deep to find out why their Entity header did not get set.

Reply:
As for now developers have to dig^2 cuz there is no explanation why their scalar values that are not stings are not passing to CURL request but they are ignoring instead. I would say scalar values should be converted here.

Reply:
No no, sorry, I was unclear. This part is great - I want to add something like this:
```php
                if (! is_scalar($value)) {
    throw new DomainException('Only scalar values are allowed');
```